File: Data Structures/Current Dump/2/2/06.DictionariesAndHashTables/DictionariesAndHashTablesHomework/03.Phonebook/Phonebook.cs
namespace _03.Phonebook
{
    using System;
    using _01.Dictionary;

    public class Phonebook
    {
        public static void Main()
        {
            var phonebook = new CustomDictionary<string, string>();

            string phoneEntry = Console.ReadLine();
            while (!phoneEntry.Equals("search"))
            {
                var args = phoneEntry.Split('-');
                string person = args[0];
                string phoneNumber = args[1];
                phonebook[person] = phoneNumber;

                phoneEntry = Console.ReadLine();
            }

            string contact = Console.ReadLine();
            while (!contact.Equals(string.Empty))
            {
                if (!phonebook.ContainsKey(contact))
                {
                    Console.WriteLine("Contact {0} does not exist.", contact);
                }
                else
                {
                    Console.WriteLine("{0} -> {1}", contact, phonebook[contact]);
                }

                contact = Console.ReadLine();
            }
        }
    }
}

File: Advanced Java/02. Sets and Maps/Exercise/src/Part2BasicMapOperations/Problem5Phonebook/Phonebook.java
package Part2BasicMapOperations.Problem5Phonebook;

import java.util.HashMap;
import java.util.Scanner;

/**
 * Created by User on 5.6.2016 г..
 */
public class Phonebook {

    public static void main(String[] args) {

        HashMap<String, String> map = new HashMap<>();
        Scanner scanner = new Scanner(System.in);
        String line = scanner.nextLine();
        while (!line.equals("search")) {
            String[] entry = line.split("-");
            map.put(entry[0],entry[1]);

            line = scanner.nextLine();
        }
        line = scanner.nextLine();
        while (!line.equals("stop")) {
            String number = map.get(line);
            if (number == null) {
                System.out.printf("Contact %s does not exist.%s", line, System.lineSeparator());
            } else {
                System.out.printf("%s -> %s%s", line, number, System.lineSeparator());
            }

            line = scanner.nextLine();
        }
    }
}
